### PR TITLE
#217 - Fix error display when user tries to add the same item to a list

### PR DESCRIPTION
**Pull request by nsanta**
Notes:
  - This change affects all components that has the button "Add to list".

TODO:
- [x] Specs

How looks now:  
add_to_list_modal_error_display

### CODE REVIEW

**Comment by ptrikutam** [comment]
@nsanta are you still working on this or is it ready for review?

**Comment by nsanta** [comment]
@ptrikutam This PR can be reviewed, just needs specs. But I want to discuss, like we said through chat, that if this is the right way to fix the issue.

File: app/views/static_pages/_connect.html.slim [hunk 1]
```diff
@@ -26,7 +26,7 @@
               #carousel-connect.carousel.slide data-ride="carousel"
                 .row
                   .carousel-inner.home-connect__feature-carousel role="listbox"
-                    - Network.includes(:users).limit(9).each_with_index do |network, index|
+                    - networks.each_with_index do |network, index|
```
Comment (ptrikutam) [inline app/views/static_pages/_connect.html.slim]: Curious why this in here? Didn't we already merge this feature?

File: app/views/static_pages/home.html.slim [hunk 2]
```diff
@@ -12,7 +12,7 @@ html
       = render 'shared/home_nav'
       = render 'static_pages/hero'
     = render 'static_pages/content'
-    = render 'static_pages/connect'
+    = render 'static_pages/connect', networks: @networks
```
Comment (ptrikutam) [inline app/views/static_pages/home.html.slim]: Same here. Didn't we already merge this feature?

File: spec/controllers/static_pages_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
```
Comment (ptrikutam) [inline spec/controllers/static_pages_controller_spec.rb]: Same with this entire file. Didn't we already merge this feature?

File: app/models/lists_item.rb [hunk 4]
```diff
@@ -4,5 +4,13 @@ class ListsItem < ApplicationRecord
 
   validates :list, presence: true
   validates :item, presence: true
-  validates_uniqueness_of :list_id, scope: [:item_id, :item_type]
+  validates(
+    :list_id,
+    uniqueness: {
+      scope: %i(item_id item_type),
+      message: lambda do |object, _data|
```
Comment (ptrikutam) [inline app/models/lists_item.rb]: Didn't know you could do this. Nice one.

File: app/views/static_pages/home.html.slim (hunk 2 above)
Comment (nsanta) [reply app/views/static_pages/home.html.slim]: No

File: spec/controllers/static_pages_controller_spec.rb (hunk 3 above)
Comment (nsanta) [reply spec/controllers/static_pages_controller_spec.rb]: No. It is part of this  PR.

File: app/views/static_pages/_connect.html.slim (hunk 1 above)
Comment (nsanta) [reply app/views/static_pages/_connect.html.slim]: I just moved the query to the controller. it is part of this PR. See StaticPagesController#home for further information.

Comment (nsanta) [reply app/views/static_pages/_connect.html.slim]: You are right. Sorry.

File: app/views/static_pages/home.html.slim (hunk 2 above)
Comment (nsanta) [reply app/views/static_pages/home.html.slim]: You are right.

**Comment by nsanta** [comment]
@ptrikutam I update the PR.

**Pull request merged**